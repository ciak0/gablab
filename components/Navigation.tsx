import { FunctionComponent } from 'react';
import Image from 'next/image';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLinkedin } from '@fortawesome/free-brands-svg-icons';
import '@fortawesome/fontawesome-svg-core/styles.css';
import logo from '../public/logo.jpg';
import { LINKED_IN_URL } from '../constants';

const Navigation: FunctionComponent = () => (
  <nav className="flex justify-between content-center sticky top-0 z-50 p-4 text-3xl bg-white shadow-sm shadow-neutral-300">
    <div className="font-extrabold text-orange-600">
      <Image
        src={logo}
        alt="Logo"
        width={18}
        height={24}
      />
      ablab.dev
    </div>
    <div>
      <a href={LINKED_IN_URL} target="_blank" rel="noreferrer">
        <FontAwesomeIcon
          size="1x"
          className=" text-blue-600"
          icon={faLinkedin}
        />
      </a>
    </div>
  </nav>
);

export default Navigation;
