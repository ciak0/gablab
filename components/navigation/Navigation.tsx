import type { FunctionComponent } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLinkedin, faBitbucket } from '@fortawesome/free-brands-svg-icons';
import '@fortawesome/fontawesome-svg-core/styles.css';
import logo from '../../public/logo.jpg';
import { LINKED_IN_URL } from '../../constants';
import LocalImage from '../local-image/LocalImage';

const Navigation: FunctionComponent = () => (
  <nav className="flex justify-between content-center sticky top-0 z-50 p-4 text-3xl bg-white shadow-sm shadow-neutral-300">
    <div className="font-extrabold text-orange-600">
      <LocalImage
        src={logo}
        alt="Logo"
        width={18}
        height={24}
      />
      ablab.dev
    </div>
    <div>
      <a href={LINKED_IN_URL} target="_blank" rel="noreferrer">
        <FontAwesomeIcon
          size="1x"
          className=" text-blue-600"
          icon={faLinkedin}
        />
      </a>
      {' '}
      <a href="https://bitbucket.org/ciak0/" target="_blank" rel="noreferrer">
        <FontAwesomeIcon
          size="1x"
          className=" text-blue-800"
          icon={faBitbucket}
        />
      </a>
    </div>
  </nav>
);

export default Navigation;
