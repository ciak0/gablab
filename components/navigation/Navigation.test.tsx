import TestRenderer from 'react-test-renderer';
import Navigation from './Navigation';

describe('components/Navigation', () => {
  it('should render correctly', () => {
    expect(TestRenderer.create(
      <Navigation />,
    ).toJSON()).toMatchSnapshot();
  });
});
