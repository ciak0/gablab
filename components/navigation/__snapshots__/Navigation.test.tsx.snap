// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation should render correctly 1`] = `
<nav
  className="
      
      sticky top-0 z-50
      flex justify-between content-center p-4 text-3xl bg-white shadow-sm shadow-neutral-300
    "
>
  <a
    className="font-extrabold text-orange-600"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="Logo"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="Logo"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="test-file-stub"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    ablab.dev
  </a>
  <div>
    <a
      href="https://www.linkedin.com/in/gabriele-ferrari-a5095b52/"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-linkedin fa-w-14 fa-1x  text-blue-600"
        data-icon="linkedin"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
     
    <a
      href="https://bitbucket.org/ciak0/"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bitbucket fa-w-16 fa-1x  text-blue-800"
        data-icon="bitbucket"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.2 32A16 16 0 0 0 6 47.8a26.35 26.35 0 0 0 .2 2.8l67.9 412.1a21.77 21.77 0 0 0 21.3 18.2h325.7a16 16 0 0 0 16-13.4L505 50.7a16 16 0 0 0-13.2-18.3 24.58 24.58 0 0 0-2.8-.2L22.2 32zm285.9 297.8h-104l-28.1-147h157.3l-25.2 147z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
</nav>
`;
